UserService/UserService.DAL/Services/EmailNotifications/EmailService.cs

```diff
@@ -89,4 +91,67 @@ public async Task SendEmailNotChangedNotificationAsync(string oldEmail, string n
 
         await _sendGridClient.SendEmailAsync(messageNew);
     }
+
+    public async Task SendOrderStatusNotificationAsync(ConsumedOrder order)
+    {
+        var subject = "";
+
+        switch (order.OrderStatus) 
+        {
+            case 0:
+                subject = "Your order was created!";
+                break;
+            case 1:
+                subject = "Your order was confirmed!";
+                break;
+            case 2:
+                subject = "Your order was completed!";
+                break;
+            case 3:
+                subject = "Your order was cancelled!";
+                break;
+        }
+
+        var plainTextBuilder = new StringBuilder();
+        var htmlBuilder = new StringBuilder();
+
+        plainTextBuilder.AppendLine($"Заказ #{order.Id}.");
+        plainTextBuilder.AppendLine($"Статус заказа: {order.OrderStatus}");
+        plainTextBuilder.AppendLine($"Общая сумма: {order.TotalPrice}");
+        plainTextBuilder.AppendLine("\nСписок товаров:");
+
+        htmlBuilder.Append("<html><body>");
+        htmlBuilder.Append("<h1>Информация о заказе</h1>");
+        htmlBuilder.Append($"<p>Заказ #{order.Id}.</p>");
+        htmlBuilder.Append($"<p>Статус заказа: {order.OrderStatus}</p>");
+        htmlBuilder.Append($"<p>Общая сумма: {order.TotalPrice}</p>");
+        htmlBuilder.Append("<h2>Список товаров:</h2>");
+        htmlBuilder.Append("<table border='1' cellpadding='5' cellspacing='0'>");
+        htmlBuilder.Append("<tr><th>Название</th><th>Количество</th><th>Цена</th><th>Скидка</th><th>Цена со скидкой</th><th>Итого</th></tr>");
+
+        foreach (var product in order.Products)
+        {
+            plainTextBuilder.AppendLine($"{product.Name} - Количество: {product.Quantity}, Цена: {product.Price}, Скидка: {product.Discount}%");
+
+            htmlBuilder.Append("<tr>");
+            htmlBuilder.Append($"<td>{product.Name}</td>");
+            htmlBuilder.Append($"<td>{product.Quantity}</td>");
+            htmlBuilder.Append($"<td>{product.Price}</td>");
+            htmlBuilder.Append($"<td>{product.Discount}%</td>");
+            htmlBuilder.Append($"<td>{product.Price * (1 - (double)product.Discount / 100)}</td>");
+            htmlBuilder.Append($"<td>{product.Price * product.Quantity * (1 - product.Discount / 100)}</td>");
+            htmlBuilder.Append("</tr>");
+        }
+
+        htmlBuilder.Append("</table>");
+        htmlBuilder.Append("</body></html>");
```

Comment:
please refactor this code, maybe you can create some factory for messages for different purposes, pass order and get result string from there. And you can use it for all methods in this service